app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt

```diff
@@ -1,197 +1,33 @@
-                                onNewGroupClicked = { newConversationViewModel.openNewGroupScreen() }
-                            )
-
-                        }
-                    )
-                }
-            )
-            composable(
-                route = NewConversationStateScreen.NEW_GROUP,
-                content = {
-                    NewGroupScreen(onBackPressed = { newConversationState.navigateBack() })
-                }
-            )
-
-        }
-    }
-}
-
-@Composable
-fun ScreenWithSearchTopBar(
-    newConversationViewModel: NewConversationViewModel,
-    searchQuery: String,
-    scrollPosition: Int,
-    content: @Composable () -> Unit
-) {
-    AppTopBarWithSearchBar(
-        scrollPosition = scrollPosition,
-        searchBarHint = stringResource(R.string.label_search_people),
-        searchQuery = searchQuery,
-        onSearchQueryChanged = { searchTerm ->
-            // when the searchTerm changes, we want to propagate it
-            // to the ViewModel, only when the searchQuery inside the ViewModel
-            // is different than searchTerm coming from the TextInputField
-            if (searchTerm != searchQuery) {
-                newConversationViewModel.search(searchTerm)
-            }
-        },
-        onSearchClicked = { newConversationViewModel.openSearchContacts() },
-        onCloseSearchClicked = {
-            newConversationViewModel.openKnownContacts()
-        },
-        appTopBar = {
-            WireCenterAlignedTopAppBar(
-                elevation = 0.dp,
-                title = stringResource(R.string.label_new_conversation),
-                navigationIconType = NavigationIconType.Close,
-                onNavigationPressed = { newConversationViewModel.close() }
-            )
-        },
-        content = content
-    )
-
-}
-
-private class NewConversationStateScreen(
-    val navController: NavHostController
-) {
-
-    var scrollPosition by mutableStateOf(0)
-        private set
-
-    fun updateScrollPosition(newScrollPosition: Int) {
-        scrollPosition = newScrollPosition
-    }
-
-    fun navigateToSearch() {
-        navController.navigate(SEARCH_PEOPLE)
-    }
-
-    fun navigateToNewGroup() {
-        navController.navigate(NEW_GROUP)
-    }
-
-    fun navigateToKnownContacts() {
-        navController.navigate(KNOWN_CONTACTS)
-    }
-
-    fun navigateBack() {
-        navController.popBackStack()
-    }
-
-
-    companion object {
-        const val KNOWN_CONTACTS = "known_contacts"
-        const val SEARCH_PEOPLE = "search_people"
-        const val NEW_GROUP = "new_group"
-
-    }
-}
-
-@OptIn(ExperimentalAnimationApi::class)
-@Composable
-private fun rememberNewConversationState(
-    navController: NavHostController = rememberNavController()
-): NewConversationStateScreen {
-    return remember {
-        NewConversationStateScreen(navController)
+    val navController = rememberNavController()
+    val listNavController = rememberNavController()
+
```

Comment:
not sure about the names here :P maybe listNavController = searchNavController and navControlller = newConversationNavController or something like that ?

Reply:
yeah I think it make more sense , will update it